weave/legacy/ops_domain/wb_domain_types.py

```diff
@@ -190,6 +190,16 @@ class ArtifactAlias(GQLBase):
 ArtifactAliasType = typing.cast(PartialObjectTypeGeneratorType, ArtifactAliasType)
 
 
+@gql_type("tag")
+class Tag(GQLBase):
+    REQUIRED_FRAGMENT = """
+        id
```

Comment:
i think we need the tag name and tagCategory name too in the `Reuired_Fragment`, no?

Reply:
Yep, pretty sure we do too 🤦‍♂️🙏

Reply:
question(learning): actually, more curious about this --

for context: I'm partly using `ArtifactAlias` as a reference here.  I totally expect some implementation differences between aliases vs tags, but is there a way I can better understand why `ArtifactAlias` only needs `id` in `REQUIRED_FRAGMENT`, even though we'd also want the `alias` field?

Reply:
Hm, good point. I'm a bit unfamiliar with weave types here. Maybe we can get some help from @tssweeney or @jwlee64 ?

Reply:
Very nice catch. you only need `id` here. if you follow the patterns for the ops as commented above, this will not be needed

```
@op(
    name="artifactTag-name",
)
def op_artifact_tag_name(tag: wdt.ArtifactTag) -> str:
```

Reply:
so on tinkering around with this a bit longer, I'll probably just stick with the following:
- keep `REQUIRED_FRAGMENT` as-is (i.e. with all fields in the required fragment: `id, name, tagCategoryName, attributes`)
- keep the related snippet in `toGqlField` as: https://github.com/wandb/weave/blob/0e4a0e8fd8f87b3bcb09ae74135cb6666e039f17/weave-js/src/core/ops/domain/gql.ts#L1123-L1124

admittedly this doesn't quite feel like the "right" way to do it, but it is somewhat motivated: in the equivalent types in FE code (for artifact/collection tags), we do expect + need all these fields to be present anyway